searx/engines/deviantart.py

```diff
@@ -7,76 +7,76 @@
  @using-api   no (TODO, rewrite to api)
  @results     HTML
  @stable      no (HTML can change)
- @parse       url, title, thumbnail_src, img_src
+ @parse       url, title, img_src
 
  @todo        rewrite to api
 """
 
-from lxml import html
-import re
 from urllib.parse import urlencode
-from searx.utils import extract_text
-
+from lxml import html
 
 # engine dependent config
 categories = ['images']
 paging = True
 time_range_support = True
 
-# search-url
-base_url = 'https://www.deviantart.com/'
-search_url = base_url + 'search?page={page}&{query}'
-time_range_url = '&order={range}'
-
-time_range_dict = {'day': 11,
-                   'week': 14,
-                   'month': 15}
+time_range_dict = dict(
+    day     = 'popular-24-hours'
+    , week  = 'popular-1-week'
+    , month = 'popular-1-month'
+    , year  = 'most-recent'
+)
 
+# search-url
+base_url = 'https://www.deviantart.com'
 
-# do search-request
 def request(query, params):
-    if params['time_range'] and params['time_range'] not in time_range_dict:
-        return params
+    """do search-request"""
 
-    params['url'] = search_url.format(page=params['pageno'],
-                                      query=urlencode({'q': query}))
+    # https://www.deviantart.com/search/deviations?page=5&q=foo
+
+    query =  dict(
```

Comment:
This indentation also seems weird. Why aren't you using the conventional indentation?